### PR TITLE
Add release notes for `v0.2.1`

### LINKED ISSUE #1908: Release v0.2.1
**What version number is to be released?**
v0.2.1

**What is the estimated date for this release?**
13.03.2023

**Release procedure**
Please refer to the [Release Procedure doc](https://github.com/VirtusLab/scala-cli/blob/main/.github/release/release-procedure.md).

**Release notes**
Please remember to create a pull request with a draft of the release notes in the [Release Notes History doc](https://github.com/VirtusLab/scala-cli/blob/main/website/docs/release_notes.md).
Please make sure the notes render correctly on [the website](https://scala-cli.virtuslab.org/docs/release_notes) - that includes swapping out GitHub-idiomatic @mentions of users, links to PRs, issues, etc.

### PR DESCRIPTION
**Pull request by MaciejG604**

### CODE REVIEW

File: website/docs/release_notes.md [hunk 1]
```diff
@@ -7,6 +7,58 @@ import ReactPlayer from 'react-player'
 
 # Release notes
 
+## [v0.2.1](https://github.com/VirtusLab/scala-cli/releases/tag/v0.2.1)
+
+### Add a guide for migrating from old `scala` runner to Scala CLI
```
Comment (Gedochao) [inline website/docs/release_notes.md]:
```suggestion
### Add a guide for migrating from the old `scala` runner to Scala CLI
```

File: website/docs/release_notes.md [hunk 2]
```diff
@@ -7,6 +7,58 @@ import ReactPlayer from 'react-player'
...
+
+As of [SIP-46](https://github.com/scala/improvement-proposals/pull/46), Scala CLI has been accepted as the new `scala`
+command. To make the transition smooth [@Gedochao](https://github.com/Gedochao)  prepared a [guide](./guides/old-runner-migration.md) highlighting
+the differences between the two runners.
```
Comment (Gedochao) [inline website/docs/release_notes.md]:
```suggestion
As of [SIP-46](https://github.com/scala/improvement-proposals/pull/46), Scala CLI has been accepted as the new `scala`
command. To make the transition smooth, we added a [guide](./guides/old-runner-migration.md) highlighting the differences between the two runners.
```
Make sure to use an absolute link to the guide on the actual release notes, as a relative link within the repo won't work there.

**Review by lwronski — APPROVED** [review]

File: website/docs/release_notes.md [hunk 3]
```diff
@@ -7,6 +7,127 @@ import ReactPlayer from 'react-player'
...
+
+As of [SIP-46](https://github.com/scala/improvement-proposals/pull/46), Scala CLI has been accepted as the new `scala`
+command. To make the transition smooth we added a [guide](docs/guides/old-runner-migration.md) highlighting
+the differences between the two runners.
+
+Added by [@Gedochao](https://github.com/Gedochao)  in [#1900](https://github.com/VirtusLab/scala-cli/pull/1900)
+
+### Improve the `publish` and `publish setup` user experience
+
+We're currently focusing on improving the experimental `publish` feature of Scala CLI and making the UX of `publish setup` + `publish` better.
```
Comment (Gedochao) [inline website/docs/release_notes.md]: Can we give examples on what was improved?

File: website/docs/release_notes.md [hunk 4]
```diff
@@ -7,6 +7,127 @@ import ReactPlayer from 'react-player'
...
+
+### Add a guide for migrating from the old `scala` runner to Scala CLI
+
+As of [SIP-46](https://github.com/scala/improvement-proposals/pull/46), Scala CLI has been accepted as the new `scala`
+command. To make the transition smooth we added a [guide](docs/guides/old-runner-migration.md) highlighting
+the differences between the two runners.
+
+Added by [@Gedochao](https://github.com/Gedochao)  in [#1900](https://github.com/VirtusLab/scala-cli/pull/1900)
+
+### Improve the `publish` and `publish setup` user experience
```
Comment (Gedochao) [inline website/docs/release_notes.md]: NIT
```suggestion
### Improve the `publish` and `publish setup` sub-commands' user experience
```

File: website/docs/release_notes.md [hunk 5]
```diff
@@ -7,6 +7,127 @@ import ReactPlayer from 'react-player'
...
+#   suppress-warning.outdated-dependencies-files   Globally suppresses warnings about outdated dependencies.
+# 
+# You are currently viewing the basic help for the config sub-command. You can view the full help by running: 
+#    scala-cli config --help-full
+# For detailed documentation refer to our website: https://scala-cli.virtuslab.org/docs/commands/misc/config
+# 
+# Config options:
+#   --unset, --remove  Remove an entry from config
+```
+Also, `config --full-help` will show the list of all keys, including the ones used in commands that require `--power`.
```
Comment (Gedochao) [inline website/docs/release_notes.md]:
```suggestion
Also, `config --full-help` will show the list of all keys.
```
`--power` is actually irrelevant here and will be addressed separately in a future PR.

File: website/docs/release_notes.md [hunk 6]
```diff
@@ -7,6 +7,127 @@ import ReactPlayer from 'react-player'
...
+  .map(_.asInstanceOf[js.Array[String]].drop(2).toSeq)
+  .getOrElse(Nil)
+val console = global.console
+console.log(argv.mkString(" "))
+```
+
+
+```bash ignore
+scala-cli . --js -- Hello World
+```
```
Comment (Gedochao) [inline website/docs/release_notes.md]: Why ignore?
Won't this work:
````suggestion
```scala title=ScalaJsArgs.scala
import scala.scalajs.js
import scala.scalajs.js.Dynamic.global

val process = global.require("process")
val argv = Option(process.argv)
  .filterNot(js.isUndefined)
  .map(_.asInstanceOf[js.Array[String]].drop(2).toSeq)
  .getOrElse(Nil)
val console = global.console
console.log(argv.mkString(" "))
```


```bash
scala-cli ScalaJsArgs.scala --js -- Hello World
```
````
?

File: website/docs/release_notes.md [hunk 7]
```diff
@@ -7,6 +7,127 @@ import ReactPlayer from 'react-player'
...
+```text
+Hello World
+```
+
+Added by [@alexarchambault](https://github.com/alexarchambault) in [#1826](https://github.com/VirtusLab/scala-cli/pull/1826)
+
+
+
+### Other changes
+* Ensure no console-syntax in reference docs and no `md` fenced blocks in `--help` by [@Gedochao](https://github.com/Gedochao)  in [#1874](https://github.com/VirtusLab/scala-cli/pull/1874)
```
Comment (Gedochao) [inline website/docs/release_notes.md]: I'd treat this as a documentation change

File: website/docs/release_notes.md [hunk 8]
```diff
@@ -7,6 +7,127 @@ import ReactPlayer from 'react-player'
...
+* Tweak error messages for running scripts without file extensions by [@Gedochao](https://github.com/Gedochao)  in [#1886](https://github.com/VirtusLab/scala-cli/pull/1886)
+* Exit with Bloop command return code if it's non-zero by [@alexarchambault](https://github.com/alexarchambault) in [#1837](https://github.com/VirtusLab/scala-cli/pull/1837)
+* bloop-rifle: increase timeout values by [@Flowdalic](https://github.com/Flowdalic) in [#1865](https://github.com/VirtusLab/scala-cli/pull/1865)
+* Suggest users to clean working directory when Nailgun server failed by [@lwronski](https://github.com/lwronski) in [#1916](https://github.com/VirtusLab/scala-cli/pull/1916)
```
Comment (Gedochao) [inline website/docs/release_notes.md:96]: I'd treat all these as fixes

File: website/docs/release_notes.md [hunk 9]
```diff
@@ -7,6 +7,127 @@ import ReactPlayer from 'react-player'
...
+### Remove unsupported kebab-case style in using directives
+
+All using directives names are now using camelCase, kebab-case is no longer available.
+
+Added by [@lwronski](https://github.com/lwronski) in [#1878](https://github.com/VirtusLab/scala-cli/pull/1878)
+
+### Add a reference for available config keys in help & docs
+
+You can now view the available config keys using `config --help`:
+```
```
Comment (Gedochao) [inline website/docs/release_notes.md]:
````suggestion
```bash
````

**Review by Gedochao — APPROVED** [review]

**Pull request merged**